src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURES_AMOUNT = FigureName.values().length;
+    private static final int MAX_SIZE = 10;
+    private static final Color DEFAULT_COLOR = Color.WHITE;
+    Random random = new Random();
+    ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        FigureName figureName = FigureName.values()[random.nextInt(FIGURES_AMOUNT)];
+        switch (figureName) {
+            case CIRCLE -> {
+                return new Circle(colorSupplier.getRandomColor(), random.nextInt(MAX_SIZE));
+            }
+            case RECTANGLE -> {
+                return new Rectangle(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_SIZE), random.nextInt(MAX_SIZE));
+            }
+            case RIGHT_TRIANGLE -> {
+                return new RightTriangle(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_SIZE), random.nextInt(MAX_SIZE));
+            }
+            case SQUARE -> {
+                return new Square(colorSupplier.getRandomColor(), random.nextInt(MAX_SIZE));
+            }
+            case ISOSCELES_TRAPEZOID -> {
+                return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_SIZE), random.nextInt(MAX_SIZE),
+                        random.nextInt(MAX_SIZE));
```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability. For example, instead of directly passing 'random.nextInt(MAX_SIZE)', assign this expression to a variable with a meaningful name and then pass that variable to the constructor.